core/include/vecmem/containers/impl/jagged_vector_buffer.ipp

```diff
@@ -70,8 +70,9 @@ jagged_vector_buffer<TYPE>::jagged_vector_buffer(
 }
 
 template <typename TYPE>
+template <typename size_t>
```

Comment:
Is there a reason this cannot be a single-layer template?

Reply:
I myself like it better this way. The buffer can be the same whether you created it with the help of `std::vector<std::size_t>` or `std::vector<unsigned int>`. Since these types are not used **anywhere** outside of the constructor(s) anyway.

Reply:
That was my first attempt but this seems not to be accepted:
```
home/gumetelo/vecmem/vecmem/core/include/vecmem/containers/impl/jagged_vector_buffer.ipp:105:1: error: too many template parameters in template redeclaration
template <typename TYPE, typename size_t>
^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/gumetelo/vecmem/vecmem/core/include/vecmem/containers/data/jagged_vector_buffer.hpp:28:1: note: previous template declaration is here
template <typename TYPE>
```

Apparently the template in the definition of the class needs to be separated from the one only for the constructor?

Reply:
Ah yes good point, that error does make sense.